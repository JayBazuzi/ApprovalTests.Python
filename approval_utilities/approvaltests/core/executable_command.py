from abc import abstractmethod, ABC


class ExecutableCommand(ABC):
    @abstractmethod
    def get_command(self) -> str:
        raise Exception("Interface member not implemented")

    @abstractmethod
    def execute_command(self, command: str) -> str:
        raise Exception("Interface member not implemented")
