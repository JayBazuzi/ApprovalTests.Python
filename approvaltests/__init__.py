"""Approvaltests module."""
from .version import *
from .utilities import *
from .approvals import *
from .asserts import *

__author__ = "Chris Lucian; Llewellyn Falco; Jim Counts; Tim Preece"
__version__ = version_number
