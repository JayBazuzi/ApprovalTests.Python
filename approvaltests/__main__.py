from .verify import verify_using_commandline_arguments

verify_using_commandline_arguments()