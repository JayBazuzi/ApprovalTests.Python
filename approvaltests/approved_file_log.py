from pathlib import Path

APPROVAL_TESTS_TEMP_DIRECTORY: str = ".approval_tests_temp"


class ApprovedFilesLog:
    @staticmethod
    def clear_log_file() -> None:
        ApprovedFilesLog.get_approved_files_log().write_text("")

    @staticmethod
    def get_approved_files_log() -> Path:
        path = Path("%s/.approved_files.log" % APPROVAL_TESTS_TEMP_DIRECTORY)
        path.parent.mkdir(parents=True, exist_ok=True)

        return path

    @staticmethod
    def log(approved_file):
        with ApprovedFilesLog.get_approved_files_log().open(mode="a") as file:
            file.write(f"{approved_file}\n")


ApprovedFilesLog.clear_log_file()
