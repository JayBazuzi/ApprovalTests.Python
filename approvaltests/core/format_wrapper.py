from typing_extensions import override
import abc
import typing


class FormatWrapper(abc.ABC):
    @abc.abstractmethod
    def wrap(self, data: typing.Any) -> typing.Any:
        pass

    @abc.abstractmethod
    def is_match(self, data: typing.Any) -> bool:
        pass


class AlwaysMatch(FormatWrapper):
    @override
    def wrap(self, data: typing.Any) -> typing.Any:
        return data

    @override
    def is_match(self, data: typing.Any) -> bool:
        return True
