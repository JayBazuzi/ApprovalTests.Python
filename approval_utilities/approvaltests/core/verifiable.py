from abc import ABC, abstractmethod
from approvaltests.core.options import Options

from approval_utilities.approvaltests.core.verify_parameters import VerifyParameters


class Verifiable(ABC):
    @abstractmethod
    def get_verify_parameters(self, options: "Options") -> VerifyParameters:
        pass
