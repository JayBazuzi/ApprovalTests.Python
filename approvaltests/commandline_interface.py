from pathlib import Path

from approvaltests import verify
from build.lib.approvaltests.commandline_interface import parse_arguments
from namer.cli_namer import CliNamer


# simplify the following lines of code from commandline_interface.py

# 1.  Modify so our future-selves would recognize it
# >>> 2.  Make a test
# 3.  Refactor parse_arguments to be part of a class (suggested by Jay)

# NEw code...
# received = args.received or stdin.read()
# Old code...
# received = args.received
# if args.received is None:
#    received = stdin.read()
# return (args.id, received)



def verify_using_commandline_arguments():
    test_id, received = parse_arguments()
    verify_with_id(received, test_id)


def verify_with_id(received, test_id):
    verify(received, namer=CliNamer(test_id=test_id))
    print(f"Test Passed: {test_id}")


if __name__ == "__main__":
    verify_using_commandline_arguments()


def list_approved_files_in_test_directory(test_dir: Path):
    return [file.name for file in test_dir.iterdir() if file.name.endswith(".approved.txt")]
