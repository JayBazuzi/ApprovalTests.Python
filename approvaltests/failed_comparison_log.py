from pathlib import Path

from approvaltests.approved_file_log import APPROVAL_TESTS_TEMP_DIRECTORY


class FailedComparisonLog:
    @staticmethod
    def clear_log_file() -> None:
        FailedComparisonLog.get_failed_comparison_log().write_text("")

    @staticmethod
    def get_failed_comparison_log() -> Path:
        path = Path("%s/.failed_comparison.log" % APPROVAL_TESTS_TEMP_DIRECTORY)
        path.parent.mkdir(parents=True, exist_ok=True)

        return path

    @staticmethod
    def log(received_file, approved_file):
        with FailedComparisonLog.get_failed_comparison_log().open(mode="a") as file:
            file.write(f"{received_file} -> {approved_file}\n")


FailedComparisonLog.clear_log_file()
