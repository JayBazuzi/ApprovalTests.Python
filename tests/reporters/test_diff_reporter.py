from typing import List

from approvaltests import verify
from approvaltests.core.reporter import Reporter
from approvaltests.reporters.diff_reporter import DiffReporter
from approvaltests.reporters.introduction_reporter import IntroductionReporter

from approvaltests.reporters.report_by_creating_diff_file import (
    ReportByCreatingDiffFile,
)
from approvaltests.core import Options


class FakeFactory:
    def get_all_reporters(self) -> List[Reporter]:
        return []


def test_fallback_reporter() -> None:
    empty_reporter_factory = FakeFactory()
    reporter = DiffReporter(empty_reporter_factory)
    # Testing implementation detail, not great but good enough for now
    assert isinstance(reporter.reporters[-1], IntroductionReporter)


def test_get_diff_file_name() -> None:
    diff_file = ReportByCreatingDiffFile.get_diff_file_name(
        "VerifyTests.test_verify_html.received.html"
    )
    assert diff_file == "VerifyTests.test_verify_html.diff"


def test_default_reporter_chain() -> None:
    # get default reporter
    reporter = Options().reporter
    printed_reporter = str(reporter)
    verify(printed_reporter)


# print the chain
# verify the chain
