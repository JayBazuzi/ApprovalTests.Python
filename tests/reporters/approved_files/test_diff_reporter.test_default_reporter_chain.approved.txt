FirstWorkingReporter({
    "name": "BeyondCompare4x64",
    "path": "C:/Program Files/Beyond Compare 4/BCompare.exe"
}, {
    "name": "BeyondCompare4",
    "path": "C:/Program Files (x86)/Beyond Compare 4/BCompare.exe"
}, {
    "name": "BeyondCompare4Mac",
    "path": "/usr/local/bin/bcompare"
}, {
    "name": "BeyondCompare3",
    "path": "C:/Program Files (x86)/Beyond Compare 3/BCompare.exe"
}, {
    "name": "AraxisMergeWin",
    "path": "C:/Program Files/Araxis/Araxis Merge/Merge.exe"
}, {
    "name": "AraxisMergeMac",
    "path": "/Applications/Araxis Merge.app/Contents/Utilities/compare"
}, {
    "name": "kdiff3",
    "path": "/usr/bin/kdiff3"
}, {
    "name": "meld",
    "path": "meld"
}, {
    "name": "DiffMerge",
    "path": "/Applications/DiffMerge.app/Contents/MacOS/DiffMerge"
}, {
    "arguments": [
        "diff"
    ],
    "name": "PyCharm",
    "path": "/Applications/PyCharm CE.app/Contents/MacOS/pycharm"
}, {
    "arguments": [
        "-d"
    ],
    "name": "ReportWithVSCode",
    "path": "code"
}, {
    "arguments": [
        "-u"
    ],
    "name": "ReportWithDiffCommandLine",
    "path": "diff"
}, PythonNativeReporter)
